src/System.Net.Sockets/tests/FunctionalTests/TcpListenerTest.cs

```diff
@@ -142,6 +142,77 @@ public void IPv6_Only_Works()
             s.Close();
         }
 
+        [Fact]
+        public async Task Accept_StartAfterStop_AcceptsSuccessfully()
+        {
+            var listener = new TcpListener(IPAddress.Loopback, 0);
+            listener.Start();
+            await VerifyAccept(listener);
+            listener.Stop();
+
+            Assert.Null(listener.Server);
+
+            listener.Start();
+            Assert.NotNull(listener.Server);
+            await VerifyAccept(listener);
+            listener.Stop();
+        }
+
+        [Fact]
+        public void ExclusiveAddressUse_ListenerNotStarted_SetAndReadSuccessfully()
+        {
+            var listener = new TcpListener(IPAddress.Loopback, 0);
+
+            Assert.False(listener.ExclusiveAddressUse);
+            listener.ExclusiveAddressUse = true;
+            Assert.True(listener.ExclusiveAddressUse);
+        }
+
+        [Fact]
+        public void ExclusiveAddressUse_SetStartListenerThenRead_ReadSuccessfully()
+        {
+            var listener = new TcpListener(IPAddress.Loopback, 0);
+
+            listener.ExclusiveAddressUse = true;
+
+            listener.Start();
+            Assert.True(listener.ExclusiveAddressUse);
+            listener.Stop();
+
+            Assert.True(listener.ExclusiveAddressUse);
+        }
+
+        [Fact]
+        public void ExclusiveAddressUse_SetStartAndStopListenerThenRead_ReadSuccessfully()
+        {
+            var listener = new TcpListener(IPAddress.Loopback, 0);
+
+            listener.Start();
+            listener.Stop();
+
+            listener.ExclusiveAddressUse = true;
+            Assert.True(listener.ExclusiveAddressUse);
+
+            listener.Start();
+            Assert.True(listener.ExclusiveAddressUse);
+            listener.Stop();
+
+            Assert.True(listener.ExclusiveAddressUse);
+        }
+
+        private static async Task VerifyAccept(TcpListener listener)
+        {
+            using (var client = new TcpClient())
+            {
+                Task connectTask = client.ConnectAsync(IPAddress.Loopback, ((IPEndPoint)listener.LocalEndpoint).Port);
+                using (Socket s = listener.AcceptSocket())
+                {
+                    Assert.False(listener.Pending());
+                }
+                await connectTask;
+            }
+        }
```

Comment:
Are you intending for this to be used in future tests, or can we move it to a local function in `Accept_StartAfterStop_AcceptsSuccessfully`?

Reply:
I intended to re-use it initially in other tests, but it was not necessary. Converted to a local function.